### PR TITLE
Schedule an emit-module-separately job even if an input is not compilable

### PR DESCRIPTION
**Pull request by xymus**
The following invocation should schedule an emit-module-separately job for src.swift and include other.dylib in the link job.

```
swiftc src.swift other.dylib -emit-library -emit-module
```

rdar://127238278

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci Please test

**Review by artemcm — APPROVED** [review]
Thank you.

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 1]
```diff
@@ -3504,6 +3504,22 @@ final class SwiftDriverTests: XCTestCase {
...
+      // Schedule an emit-module separately job even if there are non-compilable inputs.
+      var driver = try Driver(args: ["swiftc", "foo.swift", "bar.dylib", "-emit-library", "foo.dylib", "-emit-module-path", "foo.swiftmodule"],
+                              env: envVars)
+      let plannedJobs = try driver.planBuild()
+      XCTAssertEqual(plannedJobs.count, 3)
+      XCTAssertEqual(Set(plannedJobs.map { $0.kind }), Set([.compile, .emitModule, .link]))
+
+      let emitJob = try plannedJobs.findJob(.emitModule)
+      XCTAssertTrue(emitJob.commandLine.contains(try toPathOption("foo.swift")))
+      XCTAssertFalse(emitJob.commandLine.contains(try toPathOption("bar.dylib")))
```
Comment (etcwilde) [inline Tests/SwiftDriverTests/SwiftDriverTests.swift:3518]: Why is the dylib an input to the emit module job at all? This seems dangerously close to the behavior of the old driver, which would sometimes pass static archives to the frontend which ends in much sadness.

Comment (xymus) [reply Tests/SwiftDriverTests/SwiftDriverTests.swift:3518]: I can imagine it being bad, but I think this is fine. Did you notice the `XCTAssertFalse` for `bar.dylib`?

**Comment by xymus** [comment]
@swift-ci Please test

File: Tests/SwiftDriverTests/SwiftDriverTests.swift (hunk 1 above)
Comment (etcwilde) [reply Tests/SwiftDriverTests/SwiftDriverTests.swift:3518]: Oh, no, sorry, misread it as `XCTAssertTrue`, which concerned me. You're good, carry on.

**Comment by xymus** [comment]
@swift-ci Please test

**Comment by xymus** [comment]
@swift-ci Please test Windows

**Pull request merged**